frontend/src/components/RestrictedArea.js

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MIT
+// SPDX-FileCopyrightText: 2021 Dominic Heil <d.heil@campus.tu-berlin.de>, Christoph Ehm <christoph.ehmendoerfer@campus.tu-berlin.de>
+
+import React, { useEffect, useState } from "react";
+import appwriteApi from "../api/appwriteApi";
+import CenterFlexBox from "./CenterFlexBox";
+import ParagraphTypography from "./ParagraphTypography";
+import { Link } from "react-router-dom";
+import { activeTextColor } from "../assets/jss/colorPalette";
+import { arrayIntersection } from "../utils/utils";
+import { adminTeamName, partnerTeamName } from "../utils/config";
+
+const ErrorMessage = ({ children }) => (
+	<CenterFlexBox>
+		<ParagraphTypography>{children}</ParagraphTypography>
+	</CenterFlexBox>
+);
+
+const noAccessMessage = <ErrorMessage>
+	You are trying to access a user restricted area. Please <Link to="/login" style={{ textDecorationLine: "none", color: activeTextColor }}>login</Link>.
+</ErrorMessage>;
+
+const loadingMessage = <ErrorMessage>
+	Loading...
+</ErrorMessage>;
+
+const noMemberAccessMessage = <ErrorMessage>
+	You are trying to access a restricted area. If you believe you should have access to this area, please contact an admin directly.
+</ErrorMessage>;
+
+export function AdminArea({ user, children, ...options }) {
+	return <RestrictedArea user={user} teams={[adminTeamName]} {...options}>{children}</RestrictedArea>;
+}
+
+export function PartnerArea({ user, children, ...options }) {
+	return <RestrictedArea user={user} teams={[partnerTeamName, adminTeamName]} {...options}>{children}</RestrictedArea>;
+}
+
+export function LoggedInArea({ user, children, ...options }) {
+	return <RestrictedArea user={user} teams={null} {...options}>{children}</RestrictedArea>;
+}
+
+/**
+ * Displays children only if the user is member of a specific team,
+ * otherwise wrapped components will not be shown.
+ * @param children wrapped components that should only be shown to admins
+ * @param user user object as maintained by the main application component.
+ * @param teams list of team names whose members can access `children`. Any logged in user if no array.
+ * @param elseIgnore flag indicating whether error messages are suppressed.
+ * @returns {JSX.Element}
+ */
+export default function RestrictedArea({ user, teams, children, elseIgnore }) {
```

Comment:
I dont think `elseIgnore` is an intuitive name. Took me quite some time to understand the meaning of the variable. Maybe change it to something like `suppresErrorMessages` and add a small example in the JSDoc to understand what to use the variable for?